test/unit/gateways/ixopay_test.rb

```diff
@@ -2,33 +2,39 @@
 
 class IxopayTest < Test::Unit::TestCase
   def setup
-    @gateway = IxopayGateway.new(some_credential: 'login', another_credential: 'password')
+    @gateway = IxopayGateway.new(username: 'username', password: 'password', secret: 'secret')
+
+    @declined_card = credit_card('4000300011112220')
     @credit_card = credit_card
     @amount = 100
 
     @options = {
       order_id: '1',
       billing_address: address,
-      description: 'Store Purchase'
+      description: 'Store Purchase',
+      ip: '192.168.1.1'
     }
   end
 
   def test_successful_purchase
     @gateway.expects(:ssl_post).returns(successful_purchase_response)
 
     response = @gateway.purchase(@amount, @credit_card, @options)
-    assert_success response
 
-    assert_equal 'REPLACE', response.authorization
+    assert_success response
+    assert_equal 'FINISHED', response.message
+    assert_equal 'b2bef23a30b537b90fbe|20191016-b2bef23a30b537b90fbe', response.authorization
     assert response.test?
   end
 
   def test_failed_purchase
     @gateway.expects(:ssl_post).returns(failed_purchase_response)
 
-    response = @gateway.purchase(@amount, @credit_card, @options)
+    response = @gateway.purchase(@amount, @declined_card, @options)
+
     assert_failure response
-    assert_equal Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
+    assert_equal 'The transaction was declined', response.message
+    assert_equal '2003', response.error_code
   end
 
   def test_successful_authorize
```

Comment:
All of the unit tests that are not yet implemented are blank with the exception of `test_scrub`. Please consider commenting out the two assert statements in that test so that the unit tests can be run cleanly.

Reply:
Done.